src/Core/ExecutionPathTracer/ExecutionPathTracer.cs

```diff
@@ -55,15 +57,36 @@ public void OnOperationStartHandler(ICallable operation, IApplyData arguments)
         {
             this.currentDepth++;
 
+            // If `compositeTracer` is initialized, pass operations into it instead
+            if (this.compositeTracer != null)
+            {
+                this.compositeTracer.OnOperationStartHandler(operation, arguments);
+            }
```

Comment:
This is purely a stylistic thing, but if I'm understanding correctly, inside this `if` block seems like a logical place for an early `return`, since the intent is to just delegate to `compositeTracer` rather than running any code below.

Then the `else` below also becomes unnecessary.

Reply:
That's a good catch! I'll do that. :)